celery_utils/logged_task.py

```diff
@@ -36,6 +36,14 @@ def on_retry(self, exc, task_id, args, kwargs, einfo):
         """
         Capture the exception that caused the task to be retried, if any.
         """
-        super(LoggedTask, self).on_retry(exc, task_id, args, kwargs, einfo)
         if exc is not None:
             log.warning('[{}] retried due to {}'.format(task_id, einfo.traceback))
+        super(LoggedTask, self).on_retry(exc, task_id, args, kwargs, einfo)
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
```

Comment:
Ok. So, in 3.1.18, celery definitely does do this logging. See, for example, this query in splunk: index=prod-edx ("celery.worker.job" "Retry in") OR ("retried due to") (The logging happens from https://github.com/celery/celery/blob/v3.1.18/celery/worker/job.py).